### PR TITLE
Streamline android build configuration

### PR DESCRIPTION
**Pull request by ccpp**
This PR fixes some issues found while preparing a build script for updating aFreeRDP to version 2.2.0 in the F-Droid repository, described in #6565 and its linked F-Droid issues.

* It fixes the necessity of patching android configuration scripts
* It fixes forwarding of Android SDK and NDK to 3rd party builds (OpenSSL, OpenH264)
* It resolves enabling OpenH264 which needs NDK r15c while other components need newer NDKs (see https://gitlab.com/fdroid/fdroiddata/-/merge_requests/7703). This needs a separate configuration option "--openh264-ndk" to configure the r15c path to scripts/android-build-freerdp.sh

### CODE REVIEW

File: scripts/android-build-freerdp.sh [hunk 1]
```diff
@@ -37,6 +38,11 @@ do
 			WITH_OPENH264=1
 			shift
 			;;
+		--openh264-ndk)
+			shift
+			ANDROID_NDK_OPENH264=$1
```
Comment (akallabeth) [inline scripts/android-build-freerdp.sh:43]: Setting this to `ANDROID_NDK` by default would be a nice to have

File: scripts/android-build-release.conf [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+#       this is the minimal common denominator.
+#       If you require support for older NDK API levels,
+#       create seperate configurations for each NDK API level
+#       and architecture you want to support.
+WITH_JPEG=0
+WITH_OPENH264=1
+WITH_OPENSSL=1
+BUILD_DEPS=1
+DEPS_ONLY=0
+NDK_TARGET=26
```
Comment (akallabeth) [inline scripts/android-build-release.conf:17]: While NDK 26 is nicer for debugging, if you want more wide availability 21 would most likely be the best option. Don´t know current usage stats for f-droid and what the sweetspot is there.

**Review by akallabeth — APPROVED** [review]
LGTM
nice work, thank you!

**Comment by akallabeth** [comment]
@freerdp-bot test

File: scripts/android-build.conf [hunk 3]
```diff
@@ -10,15 +10,15 @@
 #       create seperate configurations for each NDK API level
 #       and architecture you want to support.
 WITH_JPEG=0
-WITH_OPENH264=0
```
Comment (akallabeth) [inline scripts/android-build.conf:13]: Please deactivate this again, the pr builder does not have the required dependenciese installed.

File: scripts/android-build-freerdp.sh (hunk 1 above)
Comment (ccpp) [reply scripts/android-build-freerdp.sh:43]: Have a look further down the same file where this is already done:

```patch
@@ -107,11 +112,19 @@ do
     fi
     if [ $WITH_OPENH264 -ne 0 ];
     then
+        if [ -z "$ANDROID_NDK_OPENH264" ]
+        then
+            echo
+            echo "Warning: Missing openh264-ndk, using $ANDROID_NDK" >&2
+            echo
+            ANDROID_NDK_OPENH264=$ANDROID_NDK
+        fi
         if [ $BUILD_DEPS -ne 0 ];
         then
             common_run bash $SCRIPT_PATH/android-build-openh264.sh \
                 --src $BUILD_SRC/openh264 --dst $BUILD_DST \

```

File: scripts/android-build.conf (hunk 3 above)
Comment (ccpp) [reply scripts/android-build.conf:13]: Would it be possible to prepare the builder for it? The effort should be manageable.
The advantage would be to have regression tests for each other pull request

File: scripts/android-build-release.conf (hunk 2 above)
Comment (ccpp) [reply scripts/android-build-release.conf]: Done. Sorry, forgot when copy-pasting

File: scripts/android-build-freerdp.sh (hunk 1 above)
Comment (akallabeth) [reply scripts/android-build-freerdp.sh:43]: oh, sorry, missed that :)

**Comment by akallabeth** [comment]
@ccpp ok, talked with our admin about that, openh264 we don´t want on our server for the time being, so please deactivate that for the time being.

File: scripts/android-build.conf (hunk 3 above)
Comment (ccpp) [reply scripts/android-build.conf]: Deactivated as discussed

**Comment by akallabeth** [comment]
@freerdp-bot test

**Review by akallabeth — APPROVED** [review]
LGTM

**Pull request merged**